create.js

```diff
@@ -46,42 +46,48 @@ module.exports = function create (path, opts, keys) {
   //   - db.getLatest
 
   db.get = function (key, cb) {
-    let isPrivate = false
-    let unbox
+    let doUnbox = false
+    let readKey
     let meta = false
     if (typeof key === 'object') {
-      isPrivate = key.private === true
-      unbox = key.unbox
+      doUnbox = key.private === true
+      readKey = key.unbox
       meta = key.meta
       key = key.id
     }
 
-    if (ref.isMsg(key)) {
-      return db.keys.get(key, function (err, data) {
-        if (err) return cb(err)
-
-        if (!isPrivate) {
-          if (meta) cb(null, { key, value: u.originalValue(data.value), timestamp: data.timestamp })
-          else cb(null, u.originalValue(data.value))
-        }
-        else {
-          const result = db._unbox(data, unbox)
-
-          if (meta) cb(null, { key, value: result.value, timestamp: result.timestamp })
-          else cb(null, result.value)
-        }
-      })
-    } else if (ref.isMsgLink(key)) {
+    if (isMsgWithQuery(key)) {
       var link = ref.parseLink(key)
       return db.get({
         id: link.link,
         private: true,
         unbox: link.query.unbox.replace(/\s/g, '+'),
         meta: link.query.meta
       }, cb)
-    } else if (Number.isInteger(key)) {
-      _get(key, cb) // seq
-    } else {
+    }
+
+    if (ref.isMsg(key)) {
+      return db.keys.get(key, function (err, data) {
+        if (err) return cb(err)
+
+        var value
+        if (!doUnbox) value = u.originalValue(data.value)
+        else {
+          if (isUnboxed(data)) value = data.value
+          else {
+            const unboxed = db._unbox(data, readKey)
+            if (unboxed) value = unboxed.value
+          }
+        }
+
+        if (!value) value = data.value // if db._unbox fails value will be undefined
+
+        if (meta) cb(null, { key, value, timestamp: data.timestamp })
+        else cb(null, value)
```

Comment:
I pulled the meta logic out into this final step instead of having it sprinkled through every branch of the if/else tree